### ISSUE TITLE
Error running AutoscalingLifecycleHook for NTH task

### ISSUE THREAD

**Issue by hakman** [kubernetes/kops#15751]
/kind bug

**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

CI

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

v1.27.4

**3. What cloud provider are you using?**

AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

```
kops create cluster --name e2e-4342699135-9b4dd.test-cncf-aws.k8s.io --cloud aws --kubernetes-version v1.27.4 --ssh-public-key /tmp/kops/e2e-4342699135-9b4dd.test-cncf-aws.k8s.io/id_ed25519.pub --set cluster.spec.nodePortAccess=0.0.0.0/0 --networking amazonvpc --set=cluster.spec.cloudProvider.aws.loadBalancerController.enabled=true --set=cluster.spec.certManager.enabled=true --master-size=t4g.medium --node-size=t4g.medium --image=ssm:/aws/service/canonical/ubuntu/server/20.04/stable/current/arm64/hvm/ebs-gp2/ami-id --zones=eu-west-1a,eu-west-1b,eu-west-1c --discovery-store=s3://k8s-kops-prow/e2e-4342699135-9b4dd.test-cncf-aws.k8s.io/discovery --admin-access 34.123.236.93/32 --master-count 1 --master-volume-size 48 --node-count 4 --node-volume-size 48
```

**5. What happened after the commands executed?**

```
W0807 05:40:43.008620    6028 executor.go:139] error running task "AutoscalingLifecycleHook/nodes-eu-west-1a-NTHLifecycleHook" (9m58s remaining to succeed): ValidationError: AutoScalingGroup name not found - No such Auto Scaling group: nodes-eu-west-1a.e2e-4342699135-9b4dd.test-cncf-aws.k8s.io
	status code: 400, request id: 538d0c0d-de15-41c0-a1d9-eec0c53d847d
W0807 05:40:43.008698    6028 executor.go:139] error running task "AutoscalingLifecycleHook/control-plane-eu-west-1a-NTHLifecycleHook" (9m58s remaining to succeed): ValidationError: AutoScalingGroup name not found - No such Auto Scaling group: control-plane-eu-west-1a.masters.e2e-4342699135-9b4dd.test-cncf-aws.k8s.io
	status code: 400, request id: 86906258-53f9-458d-ae25-28fb71a59853
W0807 05:40:43.008717    6028 executor.go:139] error running task "AutoscalingLifecycleHook/nodes-eu-west-1c-NTHLifecycleHook" (9m58s remaining to succeed): ValidationError: AutoScalingGroup name not found - No such Auto Scaling group: nodes-eu-west-1c.e2e-4342699135-9b4dd.test-cncf-aws.k8s.io
	status code: 400, request id: f45e4e6f-cc30-45d4-8ab7-61adfe53be6c
W0807 05:40:43.008729    6028 executor.go:139] error running task "AutoscalingLifecycleHook/nodes-eu-west-1b-NTHLifecycleHook" (9m58s remaining to succeed): ValidationError: AutoScalingGroup name not found - No such Auto Scaling group: nodes-eu-west-1b.e2e-4342699135-9b4dd.test-cncf-aws.k8s.io
	status code: 400, request id: 6e3cfbf3-76c6-4b33-aa82-94813445ce20
I0807 05:40:43.008751    6028 executor.go:111] Tasks: 130 done / 138 total; 8 can run
W0807 05:40:44.710695    6028 executor.go:139] error running task "AutoscalingLifecycleHook/nodes-eu-west-1a-NTHLifecycleHook" (9m56s remaining to succeed): ValidationError: AutoScalingGroup name not found - No such Auto Scaling group: nodes-eu-west-1a.e2e-4342699135-9b4dd.test-cncf-aws.k8s.io
	status code: 400, request id: 1bebbe4a-cc9d-42a5-a8e3-65970323a140
W0807 05:40:44.710757    6028 executor.go:139] error running task "AutoscalingLifecycleHook/control-plane-eu-west-1a-NTHLifecycleHook" (9m56s remaining to succeed): ValidationError: AutoScalingGroup name not found - No such Auto Scaling group: control-plane-eu-west-1a.masters.e2e-4342699135-9b4dd.test-cncf-aws.k8s.io
	status code: 400, request id: 244635bd-c8cd-4276-836e-512fc3d54deb
W0807 05:40:44.710771    6028 executor.go:139] error running task "AutoscalingLifecycleHook/nodes-eu-west-1c-NTHLifecycleHook" (9m56s remaining to succeed): ValidationError: AutoScalingGroup name not found - No such Auto Scaling group: nodes-eu-west-1c.e2e-4342699135-9b4dd.test-cncf-aws.k8s.io
	status code: 400, request id: c6a97398-cdb5-468f-86df-5d9a0295c041
W0807 05:40:44.710782    6028 executor.go:139] error running task "AutoscalingLifecycleHook/nodes-eu-west-1b-NTHLifecycleHook" (9m56s remaining to succeed): ValidationError: AutoScalingGroup name not found - No such Auto Scaling group: nodes-eu-west-1b.e2e-4342699135-9b4dd.test-cncf-aws.k8s.io
	status code: 400, request id: b05b96f4-2391-43f4-a313-fc214223f23e
I0807 05:40:44.710803    6028 executor.go:111] Tasks: 134 done / 138 total; 4 can run
I0807 05:40:45.315700    6028 executor.go:111] Tasks: 138 done / 138 total; 0 can run
I0807 05:40:48.072451    6028 dns.go:235] Pre-creating DNS records
I0807 05:40:48.385191    6028 update_cluster.go:323] Exporting kubeconfig for cluster
kOps has set your kubectl context to e2e-4342699135-9b4dd.test-cncf-aws.k8s.io
```

**6. What did you expect to happen?**

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

N/A

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

N/A

**9. Anything else do we need to know?**

See output from periodic tests:
https://testgrid.k8s.io/kops-misc#kops-aws-aws-load-balancer-controller
https://prow.k8s.io/log?container=test&id=1688424223191601152&job=e2e-kops-aws-aws-load-balancer-controller

**Comment by hakman** [comment]
@johngmyers ^^^

**Comment by hakman** [comment]
Fixed by #15846
